solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,366 @@
+  type        = string
+  default     = "private"
+  validation {
+    condition     = contains(["public", "private", "direct"], var.management_endpoint_type_for_bucket)
+    error_message = "The specified management_endpoint_type_for_bucket is not a valid selection!"
+  }
+}
+
+########################################################################################################################
+# KMS
+########################################################################################################################
+
+variable "existing_kms_instance_crn" {
+  type        = string
+  default     = null
+  description = "The CRN of the existing KMS instance (Hyper Protect Crypto Services or Key Protect)."
+}
+
+variable "existing_kms_key_crn" {
+  type        = string
+  default     = null
+  description = "The CRN of an existing KMS key to use to encrypt the Cloud Logs Object Storage bucket. If no value is set for this variable and bucket encryption is desired, specify a value for the `existing_kms_instance_crn` variable to create a key ring and key."
+}
+
+variable "kms_endpoint_type" {
+  type        = string
+  description = "The endpoint for communicating with the KMS instance. Possible values: `public`, `private.`"
+  default     = "private"
+  validation {
+    condition     = can(regex("public|private", var.kms_endpoint_type))
+    error_message = "The kms_endpoint_type value must be 'public' or 'private'."
+  }
+}
+
+variable "kms_key_ring_name" {
+  type        = string
+  default     = "cos-key-ring"
+  description = "The name for the key ring created for the Cloud Logs Object Storage bucket key. Applies only if encryption is desired and if not specifying an existing key. If a prefix input variable is specified, the prefix is added to the name in the `<prefix>-<name>` format."
+}
+
+variable "kms_key_name" {
+  type        = string
+  default     = "cos-key"
+  description = "The name for the key created for the Cloud Logs Object Storage bucket. Applies only if encryption is desired and if not specifying an existing key. If a prefix input variable is specified, the prefix is added to the name in the `<prefix>-<name>` format."
+}
+
+##############################################################################
+# Event Notification
+##############################################################################
+
+variable "existing_event_notifications_instances" {
+  type = list(object({
+    en_instance_id      = string
+    en_region           = string
+    en_integration_name = optional(string)
+    skip_en_auth_policy = optional(bool, false)
+  }))
+  default     = []
+  description = "List of Event Notifications instance details for routing critical events that occur in your IBM Cloud Logs."
```

Comment:
Complex DA inputs need a helper markdown doc link. For example, see https://github.com/terraform-ibm-modules/terraform-ibm-scc/blob/2f888db2f038d0db6cffbb228cff53fd53652f04/solutions/security-enforced/variables.tf#L85